### PR TITLE
Add simple unit tests for annotations.dart file

### PR DESCRIPTION
**Pull request by moylanm**
This PR is adding a file of two simple unit tests in order to improve code coverage (covering two simple classes within the annotations.dart file).

The issue this is fixing is the lack of complete test coverage for the annotations.dart file.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/test/foundation/annotations_test.dart [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  test('test DocumentationIcon object', () {
```
Comment (pdblasi-google) [inline packages/flutter/test/foundation/annotations_test.dart:5]: It looks like this test is testing the `url` property of the annotation and/or the constructor. Can we update the test name to reflect that?

File: packages/flutter/test/foundation/annotations_test.dart [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    const DocumentationIcon docIcon = DocumentationIcon('Test String');
+    expect(docIcon.url, contains('Test String'));
+  });
+
+  test('test Summary object', () {
```
Comment (pdblasi-google) [inline packages/flutter/test/foundation/annotations_test.dart:10]: It looks like this test is testing the `text` property of the annotation and/or the constructor. Can we update the test name to reflect that?

File: packages/flutter/test/foundation/annotations_test.dart
Comment (pdblasi-google) [inline packages/flutter/test/foundation/annotations_test.dart:1]: Can we add a similar test for the `Category` annotation? May as well test all of the annotations in this file if we're adding them.

**Review by pdblasi-google — COMMENTED** [review]
Couple of small comments, and the analysis issues need to be resolved. Thanks for contributing!

**Comment by moylanm** [comment]
I've revised the test names to reflect that we are testing the constructors of the annotations. Also, I added a test for the `Category` annotation. I hadn't included one initially because Codecov indicated that it was covered already:

Screenshot 2023-06-17 at 10 39 10

**Comment by pdblasi-google** [comment]
Awesome, thanks! Looks good overall, but I'll hold out on approval until the analysis issues are resolved. You may have to go into the check details, as the checks do some extra analysis over what's done by just the Dart analyzer.

**Comment by moylanm** [comment]
Alright, analysis checks are passing. Looks like I just needed to add the license header.

**Review by pdblasi-google — APPROVED** [review]
LGTM!

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Pull request merged**